src/java/org/apache/cassandra/service/paxos/Paxos.java

```diff
@@ -1111,7 +1111,12 @@ private static BeginResult begin(long deadline,
                     PaxosPrepare.Success success = prepare.success();
 
                     Supplier<Participants> plan = () -> success.participants;
-                    DataResolver<?, ?> resolver = new DataResolver<>(query, plan, NoopReadRepair.instance, new Dispatcher.RequestTime(query.creationTimeNanos()));
+                    long createdAtNanos = query.creationTimeNanos();
```

Comment:
bug found when enabling v2 testing.  v2 would trigger ShortReadProtection which defines the timeout based off a `deadline` which we compute based off the value published to `Dispatcher.RequestTime`, the issue is that this request doesn't have its time set, so we get `-1` which breaks the timeout!

Reply:
pushed a change to pass the `Dispatcher.RequestTime` through.  write side uses `CASRequest` and the read path just passes it through.